Mathlib/Data/Real/Basic.lean

```diff
@@ -607,3 +607,11 @@ def IsPowMul {R : Type*} [Pow R ℕ] (f : R → ℝ) :=
 def RingHom.IsBoundedWrt {α : Type*} [Ring α] {β : Type*} [Ring β] (nα : α → ℝ) (nβ : β → ℝ)
     (f : α →+* β) : Prop :=
   ∃ C : ℝ, 0 < C ∧ ∀ x : α, nβ (f x) ≤ C * nα x
+
+/-- If `x` is multiplicative with respect to `f`, then so is any `x^n`. -/
+theorem pow_mul_apply_eq_pow_mul {R : Type*} [MonoidWithZero R] (f : R → ℝ) {x : R}
+    (hx : ∀ y : R, f (x * y) = f x * f y) : ∀ (n : ℕ) (y : R), f (x ^ n * y) = f x ^ n * f y := by
```

Comment:
```suggestion
theorem pow_mul_apply_eq_pow_mul {R : Type*} [MonoidWithZero R] (f : R → ℝ) {x : R}
    (hx : ∀ y : R, f (x * y) = f x * f y) (n : ℕ) : ∀ (y : R), f (x ^ n * y) = f x ^ n * f y := by
```
then you can drop the `intro n`.

Also, this doesn't use anything about the real numbers, right? So the target of `f` could be any type with a multiplication, I guess. And then it could move to a basic algebra file.